src/core/io/src/4C_io_input_spec_builders.hpp

```diff
@@ -211,6 +229,20 @@ namespace Core::IO
       }
     };
 
+    template <typename T1, typename T2>
+    struct YamlTypeEmitter<std::pair<T1, T2>>
+    {
+      void operator()(ryml::NodeRef node, size_t*)
+      {
+        FOUR_C_ASSERT(node.is_map(), "Expected a map node.");
+        node["type"] = "pair";
+        node["first_type"] |= ryml::MAP;
+        YamlTypeEmitter<T1>{}(node["first_type"], nullptr);
```

Comment:
That's not quite correct. Depending on if pair itself is "sized" it should either be


```suggestion
        YamlTypeEmitter<T1>{}(node["first_type"], size + 1);
```

or

```suggestion
        YamlTypeEmitter<T1>{}(node["first_type"], size);
```